frontend/packages/dev-console/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx

```diff
@@ -1,26 +1,35 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { Button } from '@patternfly/react-core';
 import { impersonateStateToProps } from '@console/internal/reducers/ui';
 import { useAccessReview } from '@console/internal/components/utils';
-import { Button } from '@patternfly/react-core';
+import { AccessReviewResourceAttributes } from '@console/internal/module/k8s';
 import { rerunPipelineAndStay } from '../../../utils/pipeline-actions';
-import { PipelineModel } from '../../../models';
+import { PipelineRunModel } from '../../../models';
 import { usePipelineRunWithUserLabel } from '../../pipelineruns/triggered-by';
 import { getLatestRun, PipelineRun } from '../../../utils/pipeline-augment';
 
 type TriggerLastRunButtonProps = {
   pipelineRuns: PipelineRun[];
+  namespace: string;
   impersonate?;
 };
 
 const TriggerLastRunButton: React.FC<TriggerLastRunButtonProps> = ({
   pipelineRuns,
+  namespace,
   impersonate,
 }) => {
   const latestRun = usePipelineRunWithUserLabel(
     getLatestRun({ data: pipelineRuns }, 'startTimestamp'),
   );
-  const { label, callback, accessReview } = rerunPipelineAndStay(PipelineModel, latestRun);
+  const { label, callback } = rerunPipelineAndStay(PipelineRunModel, latestRun);
+  const accessReview: AccessReviewResourceAttributes = {
+    group: PipelineRunModel.apiGroup,
+    resource: PipelineRunModel.plural,
+    namespace,
+    verb: 'create',
+  };
   const isAllowed = useAccessReview(accessReview, impersonate);
   return (
     isAllowed && (
```

Comment:
Could you also change the button logic to be more deterministic?

```
<Button variant="secondary" onClick={callback} isDisabled={pipelineRuns.length === 0 && !!callback}>
```
Making sure we have a callback to use before enabling the button will help if there is ever an issue with the callback, we don't have a button enabled without a callback.